packages/plugin-kit/src/lib/DecodeError.ts

```diff
@@ -0,0 +1,22 @@
+import * as t from "io-ts";
+import { formatValidationErrors } from "io-ts-reporters";
+
+type DecodeErrorConstructorArgs<TInput = unknown> = {
+	input: TInput;
+	errors: t.Errors;
+};
+
+export class DecodeError<TInput = unknown> extends Error {
```

Comment:
Odd idea (optional), but i see that vite has insource testing. https://vitest.dev/guide/in-source.html

would it be worth adding an example so we know what the errors look like?

Reply:
Source testing is interesting, my opinion however is that I'd prefer to keep all test at the same location and not start spreading out tests everywhere.